cairo/ethereum/cancun/fork.cairo

```diff
@@ -195,3 +221,323 @@ func make_receipt{range_check_ptr, bitwise_ptr: BitwiseBuiltin*, keccak_ptr: Kec
+        assert tx_value = tx.value.fee_market_transaction.value.value;
+        assert blob_gas_fee = Uint(0);
+    }
+
+    if (tx.value.legacy_transaction.value != 0) {
+        assert tx_gas = tx.value.legacy_transaction.value.gas;
+        assert tx_data = tx.value.legacy_transaction.value.data;
+        assert tx_to = tx.value.legacy_transaction.value.to;
+        assert tx_value = tx.value.legacy_transaction.value.value;
+        assert blob_gas_fee = Uint(0);
+    }
+
+    if (tx.value.access_list_transaction.value != 0) {
+        assert tx_gas = tx.value.access_list_transaction.value.gas;
+        assert tx_data = tx.value.access_list_transaction.value.data;
+        assert tx_to = tx.value.access_list_transaction.value.to;
+        assert tx_value = tx.value.access_list_transaction.value.value;
+        assert blob_gas_fee = Uint(0);
+    }
+
+    let effective_gas_fee = tx_gas.value * env.value.gas_price.value;
+
+    // Calculate available gas
+    let intrinsic_cost = calculate_intrinsic_cost(tx);
+    let gas = tx_gas.value - intrinsic_cost.value;
+
+    // Increment nonce
+    increment_nonce{state=state}(sender);
+
+    // Deduct gas fee from sender
+    with_attr error_message("OverflowError") {
+        assert sender_account.value.balance.value.high = 0;
+    }
+    let sender_balance_after_gas_fee = sender_account.value.balance.value.low - effective_gas_fee -
+        blob_gas_fee.value;
+    with_attr error_message("OverflowError") {
+        assert [range_check_ptr] = sender_balance_after_gas_fee;
+        let range_check_ptr = range_check_ptr + 1;
+    }
+    tempvar sender_balance_after_gas_fee_u256 = U256(
+        new U256Struct(sender_balance_after_gas_fee, 0)
+    );
+    set_account_balance{state=state}(sender, sender_balance_after_gas_fee_u256);
+    EnvImpl.set_state{env=env}(state);
+
+    // Create preaccessed addresses and write coinbase
+    let (preaccessed_addresses_ptr) = dict_new_empty();
+    tempvar preaccessed_addresses_ptr_start = preaccessed_addresses_ptr;
+    let address = env.value.coinbase;
+    hashdict_write{poseidon_ptr=poseidon_ptr, dict_ptr=preaccessed_addresses_ptr}(
+        1, &address.value, 1
+    );
+    tempvar preaccessed_addresses = SetAddress(
+        new SetAddressStruct(
+            dict_ptr_start=cast(preaccessed_addresses_ptr_start, SetAddressDictAccess*),
+            dict_ptr=cast(preaccessed_addresses_ptr, SetAddressDictAccess*),
+        ),
+    );
+    // Create preaccessed storage keys
+    let (preaccessed_storage_keys_ptr) = dict_new_empty();
+    tempvar preaccessed_storage_keys = SetTupleAddressBytes32(
+        new SetTupleAddressBytes32Struct(
+            dict_ptr_start=cast(preaccessed_storage_keys_ptr, SetTupleAddressBytes32DictAccess*),
+            dict_ptr=cast(preaccessed_storage_keys_ptr, SetTupleAddressBytes32DictAccess*),
+        ),
+    );
+
+    if (tx.value.legacy_transaction.value == 0) {
```

Comment:
Equivalent to
https://github.com/ethereum/execution-specs/blob/d9febcb7332d62194968dcdb7775adf497c93626/src/ethereum/cancun/fork.py#L750